src/store/reducers/todo.js

```diff
@@ -1,15 +1,19 @@
-import { ADD_TASK } from "../constants/actionTypes";
+import { ADD_TASK } from "store/constants/actionTypes";
+import { generateTaskObject } from "utils/helper";
 
 const initialTodoState = {
-  tasks: [],
+  tasks: [generateTaskObject("Complete Initial setup of the Todo App")],
 };
 
 export const todoReducer = (state = initialTodoState, action) => {
   switch (action.type) {
-    case ADD_TASK:
+    case ADD_TASK: {
       return {
         ...state,
+        tasks: [{ ...action.payload }, ...state.tasks],
```

Comment:
why do we need to spread `action.payload`?

Reply:
Had a confusion regarding achieving immutability pattern in redux that' why copied `action.payload`. However understood that it is not necessary to copy action.payload to achieve immutability in redux. Thank you